tdb.go

```diff
@@ -251,24 +251,41 @@ func (db *TrailDB) GetTrailID(cookie string) (uint64, error) {
 	return uint64(trail_id), nil
 }
 
-func (db *TrailDB) Version() int {
-	return int(C.tdb_version(db.db))
+func (db *TrailDB) Version() uint64 {
+	return uint64(C.tdb_version(db.db))
 }
 
 func (db *TrailDB) Close() {
 	C.tdb_close(db.db)
 }
 
-func NewTrail(db *TrailDB, trail_id int) (*Trail, error) {
-	trail := C.tdb_cursor_new(db.db)
-	err := C.tdb_get_trail(trail, C.uint64_t(trail_id))
+func NewCursor(db *TrailDB) (*Trail, error) {
+	var err error = nil
+    var t *Trail = nil
+    trail := C.tdb_cursor_new(db.db)
+    if trail == nil{
+        err = errors.New("Could not create a new cursor (out of memory?)")
+    }else{
+        t = &Trail{db: db, trail: trail}
+	}
+	return t, err
```

Comment:
The idiomatic way (as per Go FAQs) is to separate error paths from natural paths. So something like:

```
if trail := C.tdb_cursor_new(db.db); trail != nil {
   return trail, nil
}
return nil, errors.New("...")
```